gradle-suppressible-error-prone/src/main/java/com/palantir/gradle/suppressibleerrorprone/transform/ModifyErrorProneCheckApi.java

```diff
@@ -0,0 +1,158 @@
+            return Optional.of(BugCheckerInfoVisitor::new);
+        }
+
+        if (classJarPath.equals("com/google/errorprone/VisitorState.class")
+                && getParameters().getSuppressionStage1().get()) {
+            return Optional.of(VisitorStateClassVisitor::new);
+        }
+
+        return Optional.empty();
+    }
+
+    private interface ClassTransformer {
+        Optional<byte[]> transformClass(String classJarPath, InputStream inputStream);
+    }
+
+    private void visitJar(File output, ClassTransformer classTransformer) {
+        try (ZipOutputStream zipOutputStream =
+                        new ZipOutputStream(new BufferedOutputStream(new FileOutputStream(output)));
+                JarFile jarFile = new JarFile(getInputArtifact().get().getAsFile())) {
+            jarFile.stream().forEach(jarEntry -> {
+                try {
+                    InputStream entryInputStream = jarFile.getInputStream(jarEntry);
+                    Optional<byte[]> possibleNewClassBytes =
+                            classTransformer.transformClass(jarEntry.getName(), entryInputStream);
+
+                    if (possibleNewClassBytes.isPresent()) {
+                        byte[] newClassBytes = possibleNewClassBytes.get();
+
+                        jarEntry.setSize(newClassBytes.length);
+                        jarEntry.setCompressedSize(-1);
+                        zipOutputStream.putNextEntry(jarEntry);
+                        zipOutputStream.write(newClassBytes);
+                        zipOutputStream.closeEntry();
+                    } else {
+                        zipOutputStream.putNextEntry(jarEntry);
+                        entryInputStream.transferTo(zipOutputStream);
+                        zipOutputStream.closeEntry();
+                    }
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            });
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public abstract static class Params implements TransformParameters {
+        @Input
+        public abstract Property<Boolean> getSuppressionStage1();
+
+        @Input
+        public abstract Property<String> getCacheBust();
+
+        @Nested
+        protected abstract EnvironmentVariables getEnvironmentVariables();
+
+        public Params() {
+            // When running tests, you might want to debug why the artifact transform isn't working. Let me tell you,
+            // bashing your head against the wall because *for some reason* the artifact transform isn't running at
+            // all when it should is really frustrating, only to find out hours later that it's just getting cached
+            // as the inputs have not changed. So in tests only, we set the cache busting property to some random
+            // value so the transform always happens and you can actually debug the transform.
+            getCacheBust()
```

Comment:
It's more if you've already written the gradle code, then you:
1. Run a test: it runs the transform.
2. Test fails, you rerun the test in debug mode.
3. Your breakpoint in the transform code is never hit as the transform is checked.
4. You think the transform isn't running at all.
If you change the actual code enough, it will actually rerun. But eventually it'll silently start skipping again.